### PR TITLE
Feature/pdpdevtool 2635 vs code authentication

### PR DESCRIPTION
**Pull request by jbbalbas**

### CODE REVIEW

File: packages/node-cli/src/core/CommandRegistrationService.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ module.exports = class CommandRegistrationService {
 
 		commandSetup.description(commandMetadata.description).action(async (options) => {
 			const actionResult = await executeCommandFunction(options);
-			process.exitCode = actionResult.status === ActionResult.STATUS.SUCCESS ? 0 : 1;
+			process.exitCode = actionResult.isSuccess() ? 0 : 1;
```
Comment (karl-anthony-ng) [inline packages/node-cli/src/core/CommandRegistrationService.js:53]: Is it worth it to make 0 and 1 here constants?

File: packages/node-cli/src/metadata/NodeCommandsMetadata.json [hunk 2]
```diff
@@ -60,5 +60,43 @@
                 "mandatory": false
             }
         ]
+    },
+    "account:ci": {
+        "name": "account:ci",
+        "description": "seria Set up an account to use with the SuiteCloud CLI for Node.js. This command only supports non-interactive mode.",
```
Comment (karl-anthony-ng) [inline packages/node-cli/src/metadata/NodeCommandsMetadata.json:66]: Is the description correct? There seems to be some extra text at the start.

File: packages/vscode-extension/src/types/ActionResult.d.ts [hunk 3]
```diff
@@ -10,6 +10,15 @@ export interface ActionResult<T> {
 	data: T;
 }
 
+export interface AuthenticateActionResult extends ActionResult<undefined> {
```
Comment (karl-anthony-ng) [inline packages/vscode-extension/src/types/ActionResult.d.ts:13]: It's really undefined in ActionResult<undefined>?

Comment (jbbalbas) [reply packages/vscode-extension/src/types/ActionResult.d.ts:13]: Yes, an AuthenticateActionResult doesn't have `data`, so it's an ActionResult with `data` property `undefined`

File: packages/node-cli/src/SdkExecutor.js [hunk 4]
```diff
@@ -26,97 +26,86 @@ const UTF8 = 'utf8';
...
+					if (isIntegrationMode && output.errorCode && output.errorCode === SdkErrorCodes.NO_TBA_SET_FOR_ACCOUNT) {
+						reject(NodeTranslationService.getMessage(ERRORS.SDKEXECUTOR.NO_TBA_FOR_ACCOUNT_AND_ROLE));
+					}
+					resolve(output);
+				} catch (error) {
+					reject(NodeTranslationService.getMessage(ERRORS.SDKEXECUTOR.RUNNING_COMMAND, error));
+				}
+			} else if (code !== 0) {
+				// check if the problem was due to bad Java Version
+				const javaVersionError = this._checkIfJavaVersionIssue();
```
[the remark this answers is not in the document]
Comment (jbbalbas) [reply packages/node-cli/src/SdkExecutor.js:78]: Yes, I this code is old, I only extracted it into a function, but you're right, at this point we know that java version is already valid. I'll update it

**Pull request merged**